src/language-services/analysis.ts

```diff
@@ -165,82 +162,108 @@ class DocumentAnalysis implements Analysis {
-    const maybeLine: PQP.Lexer.TLine | undefined = lexResult.lines[position.line];
+    private maybeIdentifierAt(): PQP.Language.LineToken | undefined {
+        const maybeToken: PQP.Language.LineToken | undefined = this.maybeTokenAt();
+        if (maybeToken) {
+            const token: PQP.Language.LineToken = maybeToken;
+            if (token.kind === PQP.Language.LineTokenKind.Identifier) {
+                return token;
+            }
+        }
 
-    return maybeLine !== undefined ? maybeLine.tokens : undefined;
-}
+        return undefined;
+    }
 
-function maybeTokenAt(document: TextDocument, position: Position): PQP.Language.LineToken | undefined {
-    const maybeLineTokens: ReadonlyArray<PQP.Language.LineToken> | undefined = maybeLineTokensAt(document, position);
+    private maybeLineTokensAt(): ReadonlyArray<PQP.Language.LineToken> | undefined {
+        const lexResult: PQP.Lexer.State = this.getLexerState();
+        const maybeLine: PQP.Lexer.TLine | undefined = lexResult.lines[this.position.line];
 
-    if (maybeLineTokens === undefined) {
-        return undefined;
+        return maybeLine !== undefined ? maybeLine.tokens : undefined;
     }
 
-    const lineTokens: ReadonlyArray<PQP.Language.LineToken> = maybeLineTokens;
+    private maybeTokenAt(): PQP.Language.LineToken | undefined {
+        const maybeLineTokens: ReadonlyArray<PQP.Language.LineToken> | undefined = this.maybeLineTokensAt();
 
-    for (const token of lineTokens) {
-        if (token.positionStart <= position.character && token.positionEnd >= position.character) {
-            return token;
+        if (maybeLineTokens === undefined) {
+            return undefined;
         }
-    }
 
-    // Token wasn't found - check for special case where current position is a trailing "." on an identifier
-    const currentRange: Range = {
-        start: {
-            line: position.line,
-            character: position.character - 1,
-        },
-        end: position,
-    };
+        const lineTokens: ReadonlyArray<PQP.Language.LineToken> = maybeLineTokens;
 
-    if (document.getText(currentRange) === ".") {
         for (const token of lineTokens) {
-            if (token.positionStart <= position.character - 1 && token.positionEnd >= position.character - 1) {
-                if (token.kind === PQP.Language.LineTokenKind.Identifier) {
-                    // Use this token with an adjusted position
-                    return {
-                        data: `${token.data}.`,
-                        kind: token.kind,
-                        positionStart: token.positionStart,
-                        positionEnd: token.positionEnd + 1,
-                    };
+            if (token.positionStart <= this.position.character && token.positionEnd >= this.position.character) {
+                return token;
+            }
+        }
+
+        // TODO: is this still needed with the latest parser?
```

Comment:
Just to be sure, this is a check on last lexed token, and if it's an identifier then allow a trailing comma? I think it makes more sense for consumers of the parser to allow non-standard tokens, not the parser itself.

Reply:
I forget the exact case this was handling. I'll leave in the TODO for now and clean it up when I change this to use the TriedInspection object.